spacy_llm/tests/tasks/test_entity_linker.py

```diff
@@ -0,0 +1,631 @@
+    config = Config().from_str(
+        cfg_string,
+        overrides={
+            "paths.el_nlp": str(tmp_path),
+            "paths.el_desc": str(tmp_path / "desc.csv"),
+        },
+    )
+    build_el_pipeline(nlp_path=tmp_path, desc_path=tmp_path / "desc.csv")
+    nlp = spacy.util.load_model_from_config(config, auto_fill=True)
+    assert nlp.pipe_names == ["llm"]
+
+    # also test nlp config from a dict in add_pipe
+    component_cfg = dict(config["components"]["llm"])
+    component_cfg.pop("factory")
+
+    nlp2 = spacy.blank("en")
+    nlp2.add_pipe("llm", config=component_cfg)
+    assert nlp2.pipe_names == ["llm"]
+
+
+@pytest.mark.external
+@pytest.mark.skipif(has_openai_key is False, reason="OpenAI API key not available")
+@pytest.mark.parametrize(
+    "cfg_string",
+    [
+        "zeroshot_cfg_string",
+        "fewshot_cfg_string",
+        "ext_template_cfg_string",
+    ],
+)
+def test_entity_linker_predict(cfg_string, request, tmp_path):
+    """Use OpenAI to get zero-shot LEMMA results.
+    Note that this test may fail randomly, as the LLM's output is unguaranteed to be consistent/predictable
+    """
+    cfg = request.getfixturevalue(cfg_string)
+    orig_config = Config().from_str(
+        cfg,
+        overrides={
+            "paths.el_nlp": str(tmp_path),
+            "paths.el_desc": str(tmp_path / "desc.csv"),
+        },
+    )
+    build_el_pipeline(nlp_path=tmp_path, desc_path=tmp_path / "desc.csv")
+    nlp = spacy.util.load_model_from_config(orig_config, auto_fill=True)
+    nlp.initialize(lambda: [])
+
+    text = "Alice goes to Boston to see the Boston Celtics game."
+    doc = nlp.make_doc(text)
+    doc.ents = [
+        Span(doc=doc, start=3, end=4, label="LOC"),  # Q100
+        Span(doc=doc, start=7, end=9, label="ORG"),  # Q131371
+    ]
+    doc = nlp(doc)
+    if cfg_string != "ext_template_cfg_string":
+        assert len(doc.ents) == 2
+        assert doc.ents[0].kb_id_ == "Q100"
+        assert doc.ents[1].kb_id_ == "Q131371"
+
+
+@pytest.mark.external
+@pytest.mark.parametrize(
+    "cfg_string",
+    [
+        "zeroshot_cfg_string",
+        "fewshot_cfg_string",
+    ],
+)
+def test_el_io(cfg_string, request, tmp_path):
+    cfg = request.getfixturevalue(cfg_string)
+    orig_config = Config().from_str(
+        cfg,
+        overrides={
+            "paths.el_nlp": str(tmp_path),
+            "paths.el_desc": str(tmp_path / "desc.csv"),
+        },
+    )
+    build_el_pipeline(nlp_path=tmp_path, desc_path=tmp_path / "desc.csv")
+    nlp = spacy.util.load_model_from_config(orig_config, auto_fill=True)
+    nlp.initialize(lambda: [])
+
+    assert nlp.pipe_names == ["llm"]
+    # ensure you can save a pipeline to disk and run it after loading
+    with make_tempdir() as tmpdir:
+        nlp.to_disk(tmpdir)
+        nlp2 = spacy.load(tmpdir)
+    assert nlp2.pipe_names == ["llm"]
+    nlp2.initialize(lambda: [])
+
+    text = "Alice goes to Boston to see the Boston Celtics game."
```

Comment:
What is still missing from this implementation, is a consideration for NIL cases. There's two options:
1. There are no candidates available -> we should not give the entity to the LLM but just apply NIL directly
2. There are no suitable candidates -> allow the LLM to output NIL.

For the first case, look at what happens when you change the first "Boston" here with "FooBar". The prompt will look like this:
```
ENTITIES:
- For *FooBar*:
- For *Boston Celtics*:
    Q131371. NBA team based in Boston; tied with most NBA Championships
    Q107723060. The 2021–22 Boston Celtics season was the 76th season of the franchise in the National Basketball Association (NBA). Following the Celtics' first-round exit to the Brooklyn Nets in five games from las
    Q3643001. NBA basketball team season
    Q3466394. season of National Basketball Association team the Boston Celtics
    Q3642995. NBA basketball team season
```
and the LLM will respond with 
```
For the mention *FooBar*, there is no specific entity description provided, so I chose the first entity ID <Q3643001> as it seems to be a general description of a place or venue.
```
and the first entity effictively gets tagged with Q3643001 instead of NIL.

We should fix this + test both use-cases.

Reply:
For 2, I think we should update the prompt to allow the LLM to output <NIL> specifically.

Reply:
(2) has been done in https://github.com/explosion/spacy-llm/pull/241/commits/817daabd4cbd7913e5b4ed9bac3809087a563507.
(1) requires a new code path that I'm not sure yet how to implement. Hope I'll get around to it tomorrow.

Reply:
Moved the ticket to you so you can have a look at https://github.com/explosion/spacy-llm/commit/817daabd4cbd7913e5b4ed9bac3809087a563507 meanwhile, if you have the time.

Reply:
1. is also done now. IMO this PR is functionally complete and ready for another, final review round (after which I'll write the docs PR).

Reply:
Nice work!